virtual/dist-kernel/dist-kernel-6.11.6_p2.ebuild

```diff
@@ -0,0 +1,12 @@
+# Copyright 2021-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DESCRIPTION="Virtual to depend on any Distribution Kernel"
+SLOT="0/${PVR}"
+KEYWORDS="arm64"
+
+RDEPEND="
+	~sys-kernel/asahi-kernel-${PV}
```

Comment:
Sorry, I should've observed this earlier. As-is, this isn't going to work. (This is also the kind of thing I'd expect you to have flagged as specifically needing discussion ;))

1. It causes unnecessary backtracking and a warning on skipped upgrades for any arm64 dist-kernel users.
2. It creates noise and makes it a pain to do the weekly bumps of dist-kernels.

I think we need to find another approach like adding a USE for asahi to the regular virtual although that's still going to be problematic when bumping virtual/dist-kernel.

@gentoo/dist-kernel We need to figure out how to accommodate other kernels for this, not sure how yet.

Reply:
What if we put these kernels in a separate slot of the virtual `SLOT="asahi/${PVR}"` so folks can `emerge dist-kernel:asahi` specifically.

Reply:
Oh, that could work..

Reply:
I don't think it will. Unless I'm mistaken, Portage will implicitly insist on installing the newest kernel separately, so all third-party modules will bind to the subslot of that and ignore asahi kernel.

Reply:
> I don't think it will. Unless I'm mistaken, Portage will implicitly insist on installing the newest kernel separately, so all third-party modules will bind to the subslot of that and ignore asahi kernel.

It shouldn't do that if you explicitly select `virtual/dist-kernel:asahi` right? Otherwise we might need to make some adjustments to the symlink updating logic in `kernel-install.eclass`. One solution could be to use the `linux-x.y.z/dist-kernel` identifier file to compare the `CAT/PN` of the kernel being installed with the `CAT/PN` of the kernel that the symlink currently points to.

Reply:
I think it might work with the current state of the PR, if the dep in gentoo-kernel{,-bin} (the PDEPEND) is updated to be `>=virtual/dist-kernel-${PV}:0`.

Reply:
> Unless I'm mistaken, Portage will implicitly insist on installing the newest kernel separately, so all third-party modules will bind to the subslot of that and ignore asahi kernel.

I don't see why this would happen if nothing pulls in `virtual/dist-kernel:0` to begin with, i.e. with the fixed PDEPEND.

But some clarification from @chadmed and @immolo would be welcome on exactly what scenarios they've tested with both gentoo-kernel and asahi-kernel after this PR is applied locally.

Reply:
I've had our overlaid copy of `virtual/dist-kernel` and `sys-kernel/asahi-kernel` slotted in this manner for a little while now and Portage has not tried to install newer versions from `:0` at all. Systems with the dist kernel already installed have been taken care of with a slotmove and kernel dependents now depend on `virtual/dist-kernel:asahi`. Running `emerge dist-kernel:asahi` on a fresh install results in the desired behaviour - `sys-kernel/asahi-sources` gets pulled in and Portage never tries to give you a different dist-kernel after that point.

Reply:
But the other side? If you're using gentoo-kernel, you never get anything asahi pulled in either (with the change I mentioned above needed, I assume)?